canvas_modules/common-canvas/src/object-model/api-pipeline.js

```diff
@@ -320,61 +320,72 @@ export default class APIPipeline {
 		return node;
 	}
 
-	// Returns a source node for auto completion or null if no source node can be
-	// detected. The source node is either:
+	// Returns a node for auto completion or null if no node can be
+	// detected. The node is either:
 	// 1. The selected node, if only *one* node is currently selected or
 	// 2. The most recently added node, provided it has one or more output ports or
 	// 3. The most-recent-but-one added node, provided it has one or more output ports
-	getAutoSourceNode(autoLinkOnlyFromSelNodes) {
-		var sourceNode = null;
+	//    or if autoLinkToBindingNodes is true, then it may be a binding node
+	getAutoLinkedNode(autoLinkOnlyFromSelNodes, autoLinkToBindingNodes) {
+		var autoLinkedNode = null;
 		var selectedNodes = this.objectModel.getSelectedNodes();
-
 		if (selectedNodes.length === 1 &&
-				this.isViableAutoSourceNode(selectedNodes[0])) {
-			sourceNode = selectedNodes[0];
+				this.isViableAutoLinkedNode(selectedNodes[0], autoLinkToBindingNodes)) {
+			autoLinkedNode = selectedNodes[0];
 
 		} else if (!autoLinkOnlyFromSelNodes) {
 			var nodesArray = this.getNodes();
 			if (nodesArray.length > 0) {
 				var lastNodeAdded = nodesArray[nodesArray.length - 1];
 				if (lastNodeAdded.outputs) {
-					sourceNode = lastNodeAdded;
+					autoLinkedNode = lastNodeAdded;
 				} else if (nodesArray.length > 1) {
 					var lastButOneNodeAdded = nodesArray[nodesArray.length - 2];
 					if (lastButOneNodeAdded.outputs) {
-						sourceNode = lastButOneNodeAdded;
+						autoLinkedNode = lastButOneNodeAdded;
 					}
 				}
 			}
 		}
-		return sourceNode;
+		return autoLinkedNode;
 	}
 
 	// Returns true if the node passed in is OK to be used as a source node
 	// for a node which is to be auto-added to the canvas.
-	isViableAutoSourceNode(node) {
-		return node.outputs &&
+	isViableAutoLinkedNode(node, autoLinkToBindingNodes) {
+		const viableNode = node.outputs &&
 			node.outputs.length > 0 &&
 			!CanvasUtils.isSrcCardinalityAtMax(node.outputs[0].id, node, this.getLinks());
+		if (autoLinkToBindingNodes) {
+			return viableNode || (node.inputs &&
+			node.inputs.length > 0 &&
+			!CanvasUtils.isSrcCardinalityAtMax(node.inputs[0].id, node, this.getLinks()));
+		}
+		return viableNode;
 	}
 
 	// Returns a newly created 'auto node' whose position is based on the
-	// source node (if one is provided) and the the other nodes on the canvas.
-	createAutoNode(data, sourceNode) {
+	// selected node (if one is provided) and the the other nodes on the canvas.
+	// If autoLinkToBindingNodes is true, then position the new node to the left of the selected node
+	createAutoNode(data, selectedNode, autoLinkToBindingNodes) {
```

Comment:
This isn't the necessarily the selected node -- from the caller it is the 'nodeToLinkWith` so you could call it the same here.